src/libponyc/codegen/genfun.c

```diff
@@ -621,6 +621,8 @@ void genfun_param_attrs(reach_type_t* t, reach_method_t* m, LLVMValueRef fun)
           case TK_REF:
             break;
           case TK_VAL:
+            LLVMAddAttribute(param, LLVMNoAliasAttribute);
+            // Fallthrough.
```

Comment:
I'm not super wild about this fallthrough, even with the comment.  It seems like it would be more straightforward to copy the `LLVMReadOnlyAttribute` line to this block and follow with a `break`.